keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -30,9 +31,26 @@
     private final int TOP_LEVEL_INDENTATION = 1;
     private int indentation = TOP_LEVEL_INDENTATION;
 
+    private final int vertexCount;
+    private final Map<VertexId, Integer> vertexDegrees;
+    private final double averageVertexDegree;
+
     public BayesianNetwork(Set<? extends Vertex> vertices) {
         this.vertices = ImmutableList.copyOf(vertices);
         this.vertexLabels = buildLabelMap(vertices);
+
+        this.vertexCount = vertices.size();
+        this.vertexDegrees = calculateVertexDegrees(vertices);
+        this.averageVertexDegree = calculateAverageVertexDegree(vertices);
+    }
+
+    private Map<VertexId,Integer> calculateVertexDegrees(Set<? extends Vertex> vertices) {
+        return vertices.stream().collect(Collectors.toMap(Vertex::getId, Vertex::getDegree));
+    }
+
+    private double calculateAverageVertexDegree(Set<? extends Vertex> vertices) {
+        OptionalDouble optionalDouble = vertices.stream().mapToInt(Vertex::getDegree).average();
+        return optionalDouble.isPresent() ? optionalDouble.getAsDouble() : 0.;
```

Comment:
I'd rather have a check in the constructor to see whether the set of vertices passed in isn't empty, this seems like the wrong place to check for it.